### PR TITLE
SONAR-9287 Permission is missing for some users in api/permissions/users when there's more than 20 permissions

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-db-dao/src/main/java/org/sonar/db/permission/UserPermissionDao.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import java.util.Collection;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Stream;
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/permission/UserPermissionDao.java]: MINOR Remove this unused import 'java.util.stream.Stream'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-dao/src/main/java/org/sonar/db/permission/UserPermissionMapper.java [hunk 2]
```diff
@@ -22,23 +22,20 @@
 import java.util.Collection;
 import java.util.List;
 import java.util.Set;
-import javax.annotation.Nullable;
 import org.apache.ibatis.annotations.Param;
-import org.apache.ibatis.session.RowBounds;
+import org.sonar.db.Pagination;
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/permission/UserPermissionMapper.java]: MINOR Remove this unused import 'org.sonar.db.Pagination'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-dao/src/main/resources/org/sonar/db/permission/UserPermissionMapper.xml [hunk 3]
```diff
@@ -3,55 +3,66 @@
...
+  </select>
+
+  <select id="selectUserIdsByQuery" parameterType="map" resultType="int">
+    select
+      distinct u.id, lower(u.name) as lowerName
+      <include refid="sqlQueryJoins" />
+      <where>
+        <include refid="sqlQueryFilters" />
+      </where>
+    order by lowerName asc
```
Comment (teryk) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/permission/UserPermissionMapper.xml]: why not `order by lower(u.name), u.name, ur.role` as before?

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java [hunk 4]
```diff
@@ -162,20 +163,20 @@ private static UsersWsResponse buildResponse(List<UserDto> users, List<UserPermi
   }
 
   private List<UserDto> findUsers(DbSession dbSession, PermissionQuery query) {
-    List<Integer> orderedIds = dbClient.userPermissionDao().selectUserIds(dbSession, query);
+    List<Integer> orderedIds = new ArrayList<>(dbClient.userPermissionDao().selectUserIdsByQuery(dbSession, query));
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java]: why create a new list and not immutable?

**Pull request merged**